odoo/tests/common.py

```diff
@@ -311,6 +311,14 @@ class BaseCase(TreeCase, MetaCase('DummyCase', (object,), {})):
     longMessage = True      # more verbose error message by default: https://www.odoo.com/r/Vmh
     warm = True             # False during warm-up phase (see :func:`warmup`)
 
+    def setUp(self):
+        def iap_jsonrpc(*args, **kwargs):
+            raise exceptions.AccessError("Unavailable during tests.")
+
+        iap_patch = patch('odoo.addons.iap.tools.iap_tools.iap_jsonrpc', iap_jsonrpc)
+        iap_patch.start()
```

Comment:
This call will fail if e.g. a partner or user creates a deployment without iap included for whatever reason. The `start()` call should be in a `try` block, and `addCleanup` should only be set up if the call succeeded.

Reply:
I'm not sure I understand, do you mean if IAP module isn't installed or isn't present at all ?

Reply:
The latter.

Reply:
I have a bit of trouble implementig that `try` block: `patch` will import the `target` argument when the mocked function is called, not at `patch` call. This means the exception is actually raised elsewhere.

I could try to import it just to check for its existence but that sounds pretty bad. Do you have a better idea ?

Reply:
> I have a bit of trouble implementig that `try` block: `patch` will import the `target` argument when the mocked function is called, not at `patch` call.

In my testing, `patch` performed the import during the `start` calls. Which is convenient as that pairs with the `stop` call. Do you get a different behaviour?

Reply:
No, you're right ! Sorry, I got a bit confused by the documentation. 
It works fine now, thank you.